src/clock.rs

```diff
@@ -34,7 +35,11 @@ impl Default for Error {
 /// software [`Timer`]s can be spawned from a `Clock` object.
 pub trait Clock: Sized {
     /// The type to hold the tick count
+    #[cfg(not(feature = "defmt"))]
     type T: TimeInt + Hash;
+    /// The type to hold the tick count
+    #[cfg(feature = "defmt")]
+    type T: TimeInt + Hash + defmt::Format;
```

Comment:
As a side note: I was trying to add defmt conditionally to the `TimeInt` as it is acting like more like a "trait group" where `Debug` was already included. But I had to completely copy the `TimeInt` trait with all of it's constraints, because you can't `#[cfg(...)]` single constraints.

This solution would not make it as ergonomic to use for `Format` as compared to `Debug` (if the user wants to use the `TimeInt` trait as a constraint themself that is) but it is a nice compromise and definitely useable.